IDE/QNX/example-client/client-tls.c

```diff
@@ -0,0 +1,261 @@
+{
+    int derSz, pemSz;
+    unsigned char der[4096];
+    unsigned char *pem;
+    FILE* f;
+
+    f = fopen(file, "rb");
+    if (f == NULL) {
+        fprintf(stderr, "unable to open cert file %s\n", file);
+        return -1;
+    }
+    derSz = fread(der, 1, sizeof(der), f);
+    fclose(f);
+
+    pemSz = wc_DerToPemEx(der, derSz, NULL, 0, NULL, type);
+    if (pemSz <= 0) {
+        fprintf(stderr, "issue getting pem size needed\n");
+        return -1;
+    }
+
+    pem = (unsigned char*)malloc(pemSz);
+    if (pem == NULL) {
+        fprintf(stderr, "issue malloc'ing pem size needed\n");
+        return -1;
+    }
+
+    pemSz = wc_DerToPemEx(der, derSz, pem, pemSz, NULL, type);
+    if (pemSz <= 0) {
+        fprintf(stderr, "issue %d converting der to pem\n", pemSz);
+        free(pem);
+        return -1;
+    }
+    *out = pem;
+    return pemSz;
+}
+
+
+int main(int argc, char** argv)
+{
+    int                sockfd;
+    struct sockaddr_in servAddr;
+    char               buff[256];
+    size_t             len;
+    int                ret;
+    int                port;
+
+    int pemSz;
+    unsigned char *pem;
+    FILE* f;
+
+    /* declare wolfSSL objects */
+    WOLFSSL_CTX* ctx;
+    WOLFSSL*     ssl;
+
+    /* Check for proper calling convention */
+    if (argc != 6) {
+        printf("usage: %s <port> <IPv4 address> <CA PEM> <cert DER> <key DER>\n",
+                argv[0]);
+        return 0;
+    }
+    port = atoi(argv[1]);
+
+    /* Create a socket that uses an internet IPv4 address,
+     * Sets the socket to be stream based (TCP),
+     * 0 means choose the default protocol. */
+    if ((sockfd = socket(AF_INET, SOCK_STREAM, 0)) == -1) {
+        fprintf(stderr, "ERROR: failed to create the socket\n");
+        ret = -1;
+        goto end;
+    }
+
+    /* Initialize the server address struct with zeros */
+    memset(&servAddr, 0, sizeof(servAddr));
+
+    /* Fill in the server address */
+    servAddr.sin_family = AF_INET;             /* using IPv4      */
+    servAddr.sin_port   = htons(port); /* on DEFAULT_PORT */
+
+    /* Get the server IPv4 address from the command line call */
+    if (inet_pton(AF_INET, argv[2], &servAddr.sin_addr) != 1) {
+        fprintf(stderr, "ERROR: invalid address\n");
+        ret = -1;
+        goto end;
+    }
+
+    /* Connect to the server */
+    if ((ret = connect(sockfd, (struct sockaddr*) &servAddr, sizeof(servAddr)))
+         == -1) {
+        fprintf(stderr, "ERROR: failed to connect\n");
+        goto end;
+    }
+
+    /*---------------------------------*/
+    /* Start of security */
+    /*---------------------------------*/
+    /* Initialize wolfSSL */
+    if ((ret = wolfSSL_Init()) != WOLFSSL_SUCCESS) {
+        fprintf(stderr, "ERROR: Failed to initialize the library\n");
+        goto socket_cleanup;
+    }
+
+    /* Create and initialize WOLFSSL_CTX */
+    if ((ctx = wolfSSL_CTX_new(wolfTLSv1_2_client_method())) == NULL) {
```

Comment:
Perhaps the SSLv23 or TLSv1_3 is better for the example?

Reply:
Yes, will update